test-utils/runtime-tester/fuzz/fuzz_targets/runtime_fuzzer.rs

```diff
@@ -4,22 +4,23 @@ use runtime_tester::Scenario;
 use std::fs::File;
 use std::time::Duration;
 
-fuzz_target!(|scenario: Scenario| {
-    let filename = "failed_scenario.json";
-    let runtime_stats = scenario.run();
-    if let Err(e) = runtime_stats {
-        serde_json::to_writer(&File::create(filename).unwrap(), &scenario).unwrap();
-        panic!("Bad scenario: {}, Error: {}", filename, e);
-    } else {
-        let stats = runtime_stats.unwrap();
-        for block_stats in stats.blocks_stats {
-            if block_stats.block_production_time > Duration::from_secs(1) {
-                serde_json::to_writer(&File::create(filename).unwrap(), &scenario).unwrap();
-                panic!(
-                    "Bad scenario: {}, block at height {} was produced in {:?}",
-                    filename, block_stats.height, block_stats.block_production_time
-                );
-            }
+fn do_fuzz(scenario: &Scenario) -> Result<(), String> {
+    let stats = scenario.run().result.map_err(|e| format!("{}", e))?;
+    for block_stats in stats.blocks_stats {
+        if block_stats.block_production_time > Duration::from_secs(1) {
+            return Err(format!("block at height {} was produced in {:?}",
+                               block_stats.height, block_stats.block_production_time));
         }
     }
-});
+    Ok(())
+}
+
+fn fuzz(scenario: Scenario) {
+    if let Err(err) = do_fuzz(&scenario) {
+        let file = "failed_scenario.json";
+        serde_json::to_writer(&File::create(file).unwrap(), &scenario).unwrap();
+        panic!("Bad scenario: {}, {}", file, err);
+    }
+}
```

Comment:
Sorry, I was being confusing here. What I mean is extract the function *and* move the function into `runtime-tester/src/lib.rs`, such that it is type-checked when you do `cargo check --workspace` (fuzz tagets are not part of the workspace, and that's why we are getting a compilation error). *Just* extracting the function doesn't chang much (although I must say I find it more readable when mosht of the code is outsied of macro). 

Also, I've meant to approve this originally, this is a nice-to-have, but not blocking.

Reply:
OK, I’m just gonna merge it then since this is actively breaking NayDuck.